src/library/tabledelegates/stareditor.cpp

```diff
@@ -83,31 +89,132 @@ void StarEditor::paintEvent(QPaintEvent*) {
         cg = QPalette::Inactive;
     }
 
+    painter.save();
     if (m_styleOption.state & QStyle::State_Selected) {
         painter.setBrush(m_styleOption.palette.color(cg, QPalette::HighlightedText));
     } else {
         painter.setBrush(m_styleOption.palette.color(cg, QPalette::Text));
     }
-
     m_starRating.paint(&painter, m_styleOption.rect);
-
-    // Draw a border if the color cell is selected
-    if (m_styleOption.state & QStyle::State_HasFocus) {
-        TableItemDelegate::drawBorder(&painter, m_focusBorderColor, m_styleOption.rect);
-    }
+    painter.restore();
 }
 
 bool StarEditor::eventFilter(QObject* obj, QEvent* event) {
     switch (event->type()) {
+    case QEvent::KeyPress: {
+        VERIFY_OR_DEBUG_ASSERT(m_isKeyboardEditMode) {
+            // Persistent editors (i.e. those opened using openPersistentEditor)
+            // should not receive keyboard events via their eventFilter, so we
+            // shouldn't normally arrive here - but if we do, just forward the events.
+            return false;
+        }
+
+        // Change rating when certain keys are pressed
+        // while we are in edit mode.
+        QKeyEvent* ke = static_cast<QKeyEvent*>(event);
+        int newRating = m_starRating.starCount();
+        switch (ke->key()) {
+        case Qt::Key_Up:
+        case Qt::Key_Down:
+        case Qt::Key_PageUp:
+        case Qt::Key_PageDown: {
+            // Allow handling of certain keyboard navigation events.
+            // This logic attempts to match the behavior for text editor cells.
+            return false;
+        }
+        case Qt::Key_Home:
+        case Qt::Key_End: {
+            // Only forward Home and End if the Ctrl key is pressed.
+            // This matches the behavior of normal text editor cells.
+            if (ke->modifiers() & Qt::ControlModifier) {
+                return false;
+            }
+            return true;
+        }
+        case Qt::Key_0: {
+            newRating = 0;
+            break;
+        }
+        case Qt::Key_1: {
+            newRating = 1;
+            break;
+        }
+        case Qt::Key_2: {
+            newRating = 2;
+            break;
+        }
+        case Qt::Key_3: {
+            newRating = 3;
+            break;
+        }
+        case Qt::Key_4: {
+            newRating = 4;
+            break;
+        }
+        case Qt::Key_5: {
+            newRating = 5;
+            break;
+        }
+        case Qt::Key_6: {
+            newRating = 6;
+            break;
+        }
+        case Qt::Key_7: {
+            newRating = 7;
+            break;
+        }
+        case Qt::Key_8: {
+            newRating = 8;
+            break;
+        }
+        case Qt::Key_9: {
+            newRating = 9;
+            break;
+        }
```

Comment:
Another quick note:
currently, star ratings range from 0-5, so let's use onyl those keys, and add a note to https://github.com/mixxxdj/mixxx/blob/60bdaca11e4011121895a59bbf2e05549020de63/src/track/trackrecord.h#L78

Reply:
I see your point, but I'd argue in favor of keeping it the way it currently is:
- **User Interaction:** We prevent all key press events from moving up the parent chain anyway while in edit mode (see line 201) to be consistent with the way the other editors work, so the user does not gain something from not handling the keys `6` to `9`.
https://github.com/mixxxdj/mixxx/blob/c59a384df96f5f419904e3ff4d49e668df320ccd/src/library/tabledelegates/stareditor.cpp#L192-L194
- **API Consistency:** `StarRating` has a `maxStarCount` method, so I (as a developer) would in principle expect `StarEditor` it to work nicely with every `StarRating` object I pass it, including those with max ratings other than 5, if I were to reuse it in some other place.
https://github.com/mixxxdj/mixxx/blob/3c1d44c63a2c8f080bba12885fe0771cdb96ca82/src/library/starrating.h#L23-L35
- **Code complexity:** If not for user/developer experience: What is to be gained from replacing a few lines of simple, perfectly working code with a TODO comment in an "unrelated" file?

Reply:
Sure, your points are valid.

Mine was simply about YAGNI ; )

Reply:
No offense meant or taken ; )

YAGNI is valid, but as always it's a tradeoff : )

So, ready to merge if there are no other objects?